frontend/src/selectors/index.ts

```diff
@@ -0,0 +1,29 @@
+import { createSelector } from 'reselect';
+import { AppState } from 'src/types';
+import { Category } from 'src/types/budget';
+import * as R from 'ramda';
+
+export const getCategories = (state: AppState) => state.budget.categories;
+
+export const createLoadingSelector = (actions: string[]) => (
+  state: AppState
+) => {
+  // returns true only when all actions is not loading
+  return R.any((action: string) => {
+    return R.view(R.lensPath(['ui', 'loading', action]), state);
+  }, actions);
+};
+export const createErrorMessageSelector = (actions: string[]) => (
```

Comment:
missed a line break?

Reply:
I don't know why prettier didn't add a line in between.